packages/main/src/plugin/kubernetes/contexts-manager-experimental.ts

```diff
@@ -61,29 +68,77 @@ export class ContextsManagerExperimental {
   }
 
   private async onAdd(event: DispatcherEvent): Promise<void> {
-    const previous = this.#healthCheckers.get(event.contextName);
-    previous?.dispose();
-    const newHC = new ContextHealthChecker(event.config);
-    newHC.onStateChange(this.onStateChange.bind(this));
-    this.#healthCheckers.set(event.contextName, newHC);
-    await newHC.start({ timeout: HEALTH_CHECK_TIMEOUT_MS });
+    // register and start health checker
+    const previousHealthChecker = this.#healthCheckers.get(event.contextName);
+    previousHealthChecker?.dispose();
+    const newHealthChecker = new ContextHealthChecker(event.config);
+    this.#healthCheckers.set(event.contextName, newHealthChecker);
+    newHealthChecker.onStateChange(this.onStateChange.bind(this));
+
+    newHealthChecker.onReachable(async (state: ContextHealthState) => {
+      // register and start permissions checker
+      const previousPermissionsChecker = this.#permissionsCheckers.get(state.contextName);
+      previousPermissionsChecker?.dispose();
+
+      const namespace = state.kubeConfig.getNamespace();
```

Comment:
This is not scalable, when we will address https://github.com/podman-desktop/podman-desktop/issues/8348 we cannot just update the kubeconfig, delete all informers, and restart everything with the health check etc.

I am not sure how this should be addressed, but if we can avoid rewriting everything in 2 months would be nice.

The system should from top permission to bottom

1. Can I list nodes ?
2. Can I list namespace ? Yes ? which one ?
3. Can I list everything on namespace X ? No ? then list one by one each resources ? can we simplify the number of calls ?

Reply:
Informers are started on a specific namespace (for namespaced resources), if the user changes the namespace, I don't think there is a way to keep the informers.

Health checks cannot be executed on cluster, as authentication (part of the user) is necessary. Because of this, we need to work at the level of the contexts.

I agree that we can optimize by running the health checks on merging the contexts ignoring their namespace, but I don't think the current architecture is blocking making this optimization.